cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -342,15 +324,15 @@ private void restartIfNecessary(PodRef podRef, RestartContext restartContext)
         }
 
         try {
-            RestartPlan restartPlan = restartPlan(podRef, pod, restartContext);
-            if (restartPlan.forceRestart || restartPlan.needsRestart || restartPlan.needsReconfig) {
-                if (!restartPlan.forceRestart && deferController(podRef, restartContext)) {
+            restartPlan(podRef, pod, restartContext);
```

Comment:
Myabe you should give the method some better name if it doesn't produce restart plan anymore?

Reply:
Yeah, that would work for me.

Reply:
Done